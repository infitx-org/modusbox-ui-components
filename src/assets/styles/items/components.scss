@import '../../../components/Button/Button.scss';
@import '../../../components/Checkbox/Checkbox.scss';
@import '../../../components/DatePicker/DatePicker.scss';
@import '../../../components/FileUploader/FileUploader.scss';
@import '../../../components/Heading/Heading.scss';
@import '../../../components/Icon/Icon.scss';
@import '../../../components/Menu/Menu.scss';
@import '../../../components/Modal/Modal.scss';
@import '../../../components/Modal/ModalTabsLayout/ModalTabsLayout.scss';
@import '../../../components/RadioGroup/RadioGroup.scss';
@import '../../../components/ScrollBox/ScrollBox.scss';
@import '../../../components/ScrollBox/ScrollBar.scss';
@import '../../../components/Select/Select.scss';
@import '../../../components/Select/Options.scss';
@import '../../../components/Spinner/Spinner.scss';
@import '../../../components/Tabs/Tabs.scss';
@import '../../../components/TextField/TextField.scss';
@import '../../../components/Toast/Toast.scss';
@import '../../../components/Tooltip/Tooltip.scss';
@import '../../../components/Tooltip/TooltipContent.scss';
