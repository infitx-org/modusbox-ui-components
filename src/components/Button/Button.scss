@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/shadow.scss';

.input-button__input {
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  display: inline-block;
  flex: 0 0 auto;
  white-space: nowrap;
  user-select: none;
  text-decoration: none;
  transition: background 0.05s ease-out, margin 0.01s linear;
  font-style: normal;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  border: 1px solid;
  align-items: center;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;
  border-bottom-width: 2px;
  &:not(.noFill) {
    border-bottom-color: rgba(0, 0, 0, 0.2);
  }
  &:hover,
  &:focus {
    border-bottom-width: 2px;
    border-bottom-color: rgba(0, 0, 0, 0.2);
    &:active {
      border-bottom-color: transparent;
    }
  }

  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}

.input-button__content {
  display: flex;
  align-items: center;
  justify-content: center;
}
.input-button__icon {
  display: flex;
  margin-right: 5px;
  margin-left: 0px;
  fill: inherit;
  stroke: inherit;
}

.input-button__mb-input--small {
  padding: 6px 10px;
  font-size: 12px;
}
.input-button__mb-input--medium {
  padding: 8px 13px;
  font-size: 13px;
}
.input-button__mb-input--large {
  padding: 10px 16px;
  font-size: 14px;
}

.input-button__mb-input--primary {
  background-color: $color__main-primary;
  border-color: $color__main-primary;

  &.noFill {
    background-color: transparent;
    color: $color__main-primary;
    fill: $color__main-primary;
    stroke: $color__main-primary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--primary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-primary--active;
    border-color: $color__main-primary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-button__mb-input--secondary {
  background-color: $color__main-secondary;
  border-color: $color__main-secondary;

  &.noFill {
    background: transparent;
    color: $color__main-secondary;
    fill: $color__main-secondary;
    stroke: $color__main-secondary;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--secondary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-secondary--active;
    border-color: $color__main-secondary--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--success {
  background-color: $color__main-success;
  border-color: $color__main-success;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-success;
    fill: $color__main-success;
    stroke: $color__main-success;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--success;
  }

  &:hover,
  &:focus {
    background-color: $color__main-success--active;
    border-color: $color__main-success--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}
.input-button__mb-input--danger {
  background-color: $color__main-danger;
  border-color: $color__main-danger;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background: transparent;
    color: $color__main-danger;
    fill: $color__main-danger;
    stroke: $color__main-danger;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--danger;
  }

  &:hover,
  &:focus {
    background-color: $color__main-danger--active;
    border-color: $color__main-danger--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--warning {
  background-color: $color__main-warning;
  border-color: $color__main-warning;

  &.noFill {
    background-color: transparent;
    color: $color__main-warning;
    fill: $color__main-warning;
    stroke: $color__main-warning;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--warning;
  }

  &:hover,
  &:focus {
    background-color: $color__main-warning--active;
    border-color: $color__main-warning--active;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--dark {
  background-color: $color__alpha-5;
  border-color: transparent;
  color: $color__white;
  fill: $color__white;
  stroke: $color__white;

  &.noFill {
    background-color: transparent;
    border-color: $color__alpha-5;
    color: $color__alpha-4;
    fill: $color__alpha-4;
    stroke: $color__alpha-4;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--dark;
  }

  &:hover,
  &:focus {
    background-color: $color__alpha-4;
    border-color: transparent;

    &.noFill {
      color: $color__white;
      fill: $color__white;
      stroke: $color__white;
    }
  }
}

.input-button__mb-input--light {
  background-color: $color__white;
  border-color: transparent;
  color: $color__gray-5;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    background-color: transparent;
    border-color: $color__white;
    color: $color__white;
    fill: $color__white;
    stroke: $color__white;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--light;
  }

  &:hover,
  &:focus {
    background-color: $color__white;
    border-color: transparent;
    color: $color__gray-4;

    &.noFill {
      color: $color__gray-5;
      fill: $color__gray-5;
      stroke: $color__gray-5;
    }
  }
}



.input-button__mb-input--disabled {
  background-color: $color__main-tertiary;
  border-color: $color__main-tertiary;
  color: $color__dark-gray-1;
  fill: $color__gray-5;
  stroke: $color__gray-5;

  &.noFill {
    border-color: $color__main-tertiary;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--tertiary;
  }

  &:hover,
  &:focus {
    background-color: $color__main-tertiary--active;
    border-color: $color__main-tertiary--active;
  }
}
.input-button__mb-input--tertiary {
  background-color: $color__gray-4;
  border-color: $color__gray-4;
  color: $color__dark-gray-2;
  fill: $color__dark-gray-2;
  stroke: $color__dark-gray-2;

  &.noFill {
    border-color: $color__gray-4;
    background: transparent;
    color: $color__gray-5;
    fill: $color__gray-5;
    stroke: $color__gray-5;
  }

  &:active,
  &:focus {
    box-shadow: $shadow-size $shadow-color--tertiary;
  }

  &:hover,
  &:focus {
    color: $color__dark-gray-2;
    fill: $color__dark-gray-2;
    stroke: $color__dark-gray-2;
    background-color: $color__gray-3;
    border-color: $color__gray-3;
  }
}

.input-button__mb-input--primary,
.input-button__mb-input--secondary,
.input-button__mb-input--tertiary,
.input-button__mb-input--danger,
.input-button__mb-input--warning,
.input-button__mb-input--dark,
.input-button__mb-input--light {
  &:hover {
    &.noFill {
      &:active {
        color: $color__alpha-3;
        fill: $color__alpha-3;
        stroke: $color__alpha-3;
      }
    }
  }
  &:active {
    color: $color__alpha-3;
    fill: $color__alpha-3;
    stroke: $color__alpha-3;
  }
}
