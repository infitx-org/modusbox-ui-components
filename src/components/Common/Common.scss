@import '../../assets/styles/vars/colors.scss';
.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 10px;
  box-shadow: 0px 1px 2px $color__alpha-2;
}
.validation__message {
  display: flex;
  align-items: center;
  white-space: normal;
  word-break: break-all;
  font-size: 12px;
  color: $color__dark-gray-1;
  fill: $color__main-danger;
}
.validation__message--active {
  text-decoration: line-through;
  color: $color__gray-5;
  fill: $color__main-success;
}
.validation__message-icon {
}
.validation__message-text {
  margin-left: 3px;
}
