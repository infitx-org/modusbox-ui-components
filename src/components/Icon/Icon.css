.element-icon{
	flex: 0 0 auto;
	&.spin{
		animation: spinIcon 1s linear infinite;
	}
}

@keyframes spinIcon{
	0%{
		transform: rotate(0deg);
	}
	100%{
		transform: rotate(360deg);
	}
}