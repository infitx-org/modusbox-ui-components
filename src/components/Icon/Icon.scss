@import '../../assets/styles/vars/colors.scss';

.element-icon {
  display: flex;
  flex: 0 0 auto;
}
.element-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
