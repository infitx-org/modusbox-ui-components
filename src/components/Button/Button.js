import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import * as utils from '../../utils/common';

import Icon, { iconSizes } from '../Icon';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

class Button extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.testKey = this.testKey.bind(this);
  }

  componentWillReceiveProps() {}

  onClick(e) {
    if (this.props.disabled) return;
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  testKey(e) {
    if (e.nativeEvent.keyCode === 9) {
      e.preventDefault();
      utils.focusNextFocusableElement(this.input, !e.nativeEvent.shiftKey);
      return;
    }

    if (e.nativeEvent.keyCode === 13) {
      e.preventDefault();
      this.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      kind,
      label,
      icon,
      noFill,
      disabled,
      pending,
      tooltip,
    } = this.props;
    const isDisabledOrPending = disabled === true || pending === true;
    const iconSize = iconSizes[size];
    const classNames = utils.composeClassNames([
      className,
      'mb-input',
      'input-button__input',
      size === 's' && 'input-button__mb-input--small',
      size === 'm' && 'input-button__mb-input--medium',
      size === 'l' && 'input-button__mb-input--large',
      kind === 'primary' && 'input-button__mb-input--primary',
      kind === 'secondary' && 'input-button__mb-input--secondary',
      kind === 'tertiary' && 'input-button__mb-input--tertiary',
      kind === 'success' && 'input-button__mb-input--success',
      kind === 'danger' && 'input-button__mb-input--danger',
      kind === 'warning' && 'input-button__mb-input--warning',
      kind === 'dark' && 'input-button__mb-input--dark',
      kind === 'light' && 'input-button__mb-input--light',
      isDisabledOrPending && 'mb-input--disabled input-button__mb-input--disabled',
      pending && 'mb-input--pending input-button__mb-input--pending',
      noFill && 'noFill',
    ]);

    const button = (
      <button
        ref={input => {
          this.input = input;
        }}
        id={id}
        style={style}
        className={classNames}
        onKeyDown={this.testKey}
        onClick={this.onClick}
        disabled={isDisabledOrPending}
        label={label}
        kind={kind}
      >
        <div className="input-button__content">
          {(pending || icon) && (
            <div className="input-button__icon">
              {pending ? (
                <Spinner color="inherit" size={iconSize} />
              ) : (
                <Icon name={icon} stroke="none" spin={pending} size={iconSize} />
              )}
            </div>
          )}
          {label && <span>{label}</span>}
        </div>
      </button>
    );

    if (tooltip) {
      return (
        <Tooltip label={tooltip} position="top">
          {button}
        </Tooltip>
      );
    }
    return button;
  }
}
Button.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  style: PropTypes.shape(),
  kind: PropTypes.oneOf([
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  size: PropTypes.oneOf([
    's',
    'm',
    'l',
  ]),
  label: PropTypes.string,
  icon: PropTypes.string,
  noFill: PropTypes.bool,
  disabled: PropTypes.bool,
  pending: PropTypes.bool,
  onClick: PropTypes.func,
  tooltip: PropTypes.string,
};
Button.defaultProps = {
  className: undefined,
  id: undefined,
  style: undefined,
  kind: 'primary',
  size: 'l',
  label: undefined,
  icon: undefined,
  noFill: false,
  disabled: false,
  pending: false,
  onClick: undefined,
  tooltip: undefined,
};

export default Button;
