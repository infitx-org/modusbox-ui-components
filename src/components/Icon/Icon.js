import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import * as utils from '../../utils/common';

import Tooltip from '../Tooltip';

import './Icon.scss';

const iconSizes = {
  s: 12,
  m: 14,
  l: 16,
};

class Icon extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      name,
      fill,
      stroke,
      spin,
      tooltip,
      tooltipPosition,
    } = this.props;
    const svgStyle = {
      height: `${size}px`,
      width: `${size}px`,
      fill,
      stroke,
      ...style,
    };
    const componentClassName = utils.composeClassNames([
      'element-icon',
      spin && 'element-icon--spin',
      className,
    ]);
    const svg = (
      <svg id={id} style={svgStyle} onClick={this.onClick} className={componentClassName}>
        <use xlinkHref={`#${name}`} />
      </svg>
    );
    if (tooltip) {
      return (
        <Tooltip label={tooltip} position={tooltipPosition}>
          {svg}
        </Tooltip>
      );
    }
    return svg;
  }
}

Icon.propTypes = {
  id: PropTypes.string,
  style: PropTypes.shape(),
  className: PropTypes.string,
  size: PropTypes.number,
  name: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  spin: PropTypes.bool,
  onClick: PropTypes.func,
  tooltip: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
};
Icon.defaultProps = {
  id: undefined,
  style: undefined,
  className: undefined,
  size: 20,
  name: undefined,
  fill: undefined,
  stroke: undefined,
  spin: false,
  onClick: undefined,
  tooltip: undefined,
  tooltipPosition: undefined,
};

export default Icon;
export { iconSizes };
