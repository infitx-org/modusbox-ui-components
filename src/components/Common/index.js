import React from 'react';
import PropTypes from 'prop-types';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, active, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    active && 'mb-input__inner-button--active',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  active: false,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ValidationMessage = ({ text, active }) => (
  <li className={`validation__message ${active ? 'validation__message--active' : ''}`}>
    <Icon name={active ? 'check-small' : 'close-small'} size={12} />
    <span className="validation__message-text">{text}</span>
  </li>
);

ValidationMessage.propTypes = {
  text: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  text: undefined,
  active: false,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map((message, i) => (
      <ValidationMessage key={i.toString()} text={message.text} active={message.active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.oneOfType([PropTypes.arrayOf(ValidationMessage.defaultProps)]),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ messages, forceTooltipVisibility }) => (
  <Tooltip
    position="right"
    kind="error"
    custom
    content={<ValidationMessages messages={messages} />}
    forceVisibility={forceTooltipVisibility}
  >
    <Icon size={16} name="warning-sign" />
  </Tooltip>
);
const Validation = ({ active, className, messages }) => {
  // Validation Icon with custom tooltip
  const invalidIconClassName = utils.composeClassNames([
    'mb-input__inner-icon',
    'mb-input__inner-icon--invalid',
    className,
  ]);
  return (
    <div className={invalidIconClassName}>
      <InvalidIcon messages={messages} forceTooltipVisibility={active} />
    </div>
  );
};

export { Loader, Placeholder, InnerButton, Validation };
