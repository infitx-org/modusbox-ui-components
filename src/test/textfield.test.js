import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TextField from '../components/TextField';
import Button from '../components/Button';
import Spinner from '../components/Spinner';
import Icon from '../components/Icon';
import { Loader, Placeholder, InnerButton, Validation } from '../components/Common';

it('renders the textfield', () => {
  const wrapper = shallow(<TextField />);
  expect(wrapper.find('div.input-textfield')).toHaveLength(1);
});

it('renders the textfield value', () => {
  const wrapper = shallow(<TextField value="test-value" />);
  expect(wrapper.find('input').prop('value')).toBe('test-value');
});

it('renders the placeholder', () => {
  const wrapper = shallow(<TextField placeholder="test-textfield" />);
  expect(wrapper.find(Placeholder)).toHaveLength(1);
  expect(wrapper.find(Placeholder).prop('label')).toBe('test-textfield');
});

it('applies the prop className', () => {
  const wrapper = shallow(<TextField className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(<TextField id="testTextFieldId" />);
  expect(wrapper.find('#testTextFieldId')).toHaveLength(1);
});

it('renders the disabled state', () => {
  const wrapper = shallow(<TextField disabled />);
  expect(wrapper.find('input')).toHaveLength(1);
  expect(wrapper.find('input').prop('disabled')).toBe(true);
});

it('renders the icon', () => {
  const wrapper = mount(<TextField icon="deploy" />);
  expect(wrapper.find(Icon).prop('name')).toBe('deploy');
});

it('renders the pending state', () => {
  const wrapper = shallow(<TextField pending />);
  expect(wrapper.find(Loader)).toHaveLength(1);
});

it('renders the invalid state', () => {
  const wrapper = shallow(<TextField invalid />);
  expect(wrapper.find(Validation)).toHaveLength(1);
});

it('renders the required state', () => {
  const wrapper = shallow(<TextField required />);
  expect(wrapper.find('.mb-input--required')).toHaveLength(1);
});

it('renders the password kind and clicks the toggle icon', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField type="password" />);
  expect(wrapper.find(Icon)).toHaveLength(1);
  expect(wrapper.find(Icon).prop('name')).toBe('toggle-visible');
  expect(wrapper.find('input').prop('type')).toBe('password');
  wrapper.find(Icon).simulate('click');
  expect(wrapper.find(Icon).prop('name')).toBe('toggle-invisible');
  expect(wrapper.find('input').prop('type')).toBe('text');
});

// Inner button

it('renders the inner button', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onButtonClick={mockEvent} buttonText="inner-button-text" />);
  expect(wrapper.find(Button).text()).toBe('inner-button-text');
});

it('renders the inner button disabled', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <TextField onButtonClick={mockEvent} buttonText="inner-button-text" buttonDisabled />
  );
  expect(wrapper.find(Button).prop('disabled')).toBe(true);
});

it('renders the inner button with kind', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(
    <TextField onButtonClick={mockEvent} buttonText="inner-button-text" buttonKind="secondary" />
  );
  expect(wrapper.find(Button).prop('kind')).toBe('secondary');
});

it('renders and clicks the inner button', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onButtonClick={mockEvent} buttonText="inner-button-text" />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find(Button).simulate('click');
  expect(mockEvent).toHaveBeenCalled();
});

// Events

it('triggers onfocus when focused', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onFocus={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('focus');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onBlur when blurred', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onBlur={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('blur');
  expect(mockEvent).toHaveBeenCalled();
});

it('triggers onChange when changed', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<TextField onChange={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('change');
  expect(mockEvent).toHaveBeenCalled();
});

// Snapshot

it('renders the textfield correctly when multiple props are set', () => {
  const wrapper = shallow(<TextField value="test-value" id="test-id" />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
