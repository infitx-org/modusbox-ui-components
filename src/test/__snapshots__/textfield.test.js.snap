// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the textfield correctly when multiple props are set 1`] = `
<div
  className="input-textfield mb-input__box"
  style={Object {}}
>
  <div
    className="input-textfield__component mb-input mb-input__borders mb-input__background"
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="mb-input__content input-textfield__content"
    >
      <input
        autoComplete="off"
        autoFocus={false}
        className="mb-input__input input-textfield__value"
        disabled={false}
        id="test-id"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        type="text"
        value="test-value"
      />
    </div>
  </div>
</div>
`;
